### PR TITLE
#310 Use CreateView instead of function views

### PR DESCRIPTION
**Pull request by vryazanov**
#310 
* Use generic view instead of function views

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#384](https://github.com/timelesslounge/timelessis/issues/384) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#384](https://github.com/timelesslounge/timelessis/issues/384) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/e6de84c7-0b2c-45f3-ba94-b347436eef54); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/restaurants/table_shapes/views.py [hunk 1]
```diff
@@ -32,24 +34,15 @@ def list():
...
-        """
-        Uncomment after correction
-        form.save()
-        """
-        return redirect(url_for("table_shape.list"))
-    return render_template(
-        "restaurants/table_shapes/create_edit.html", form=form)
+class Create(views.CreateView):
+    """
+    @todo #162:30min This form currenly cannot save pictures. Need have a
```
Comment (amihaiemil) [inline timeless/restaurants/table_shapes/views.py]: @vryazanov why can't it process images, what exactly is the exception? Furthermore, I don't think the Form has to process images, but only paths to them? 

It depends how you wanna do it. I would personally only save the path to the image in the DB (and store the Image in some other way, not through this form). For instance, when you upload files manually to Github, it sends them to AWS S3 behind the scenes and Github itself only stores paths.

File: timeless/templates/restaurants/floors/create_edit.html [hunk 2]
```diff
@@ -12,6 +12,6 @@ <h1>{% block title %}Floors management - {% if action == 'create' %} Create {% e
     <label for="description">Description</label>
     <textarea name="description" id="description"></textarea>
     <input type="submit" value="{% if action == 'create' %} Create {% else %} Update {% endif %}">
-    <a class="action" href="{{ url_for('floor.list_floors') }}">Cancel</a>
+    <a class="action" href="{{ url_for('floor.list') }}">Cancel</a>
   </form>
 {% endblock %}
```
Comment (amihaiemil) [inline timeless/templates/restaurants/floors/create_edit.html]: @vryazanov new line here

File: timeless/restaurants/table_shapes/views.py (hunk 1 above)
Comment (vryazanov) [reply timeless/restaurants/table_shapes/views.py]: @amihaiemil I absolutely agree with you. There is should be only path in the db, but the logic of receiving multipart/formdata should be still finished.
We just need to use the code from https://flask-wtf.readthedocs.io/en/stable/form.html#module-flask_wtf.file and check that it works.
btw it's not a new puzzle, it already was here

File: tests/integration/it_floors_test.py [hunk 3]
```diff
@@ -23,10 +22,12 @@ def test_login_required(client, path):
                                                    _external=True)
 
 
-@pytest.mark.skip(reason="auth.login() is not yet implemented")
-def test_create(client, auth):
-    auth.login()
```
[the remark this answers is not in the document]
Comment (vryazanov) [reply tests/integration/it_floors_test.py]: @amihaiemil it's a good question. I didn't touch permissions in this PR at all. so it means it didnt work before. I'll leave a puzzle for checking permissions.

**Comment by vryazanov** [comment]
> @vryazanov looks ok, see questions. Again, the PR is rather big... normally I would reject it (it's also not ok for you to work more than what you are paid for), but I understand the Project is in quite a hurry to meet deadlines.

@amihaiemil i've updated the code, please have a look. actually I didn't spend on implementation more them 30min. but yes, but recently I faced with merge conflicts and it took sometime %)  I'm trying to contribute less and get more money, but as you can see without success. hope the next my PR will be shorter :D

**Comment by amihaiemil** [comment]
@vryazanov thanks! :)

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#384](https://github.com/timelesslounge/timelessis/issues/384) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
The job [#384](https://github.com/timelesslounge/timelessis/issues/384) is now out of scope

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**